manifests/instance.pp

```diff
@@ -210,9 +216,23 @@
     $redis_file_name_orig = $config_file_orig
     $redis_file_name      = $config_file
   } else {
-    $redis_config_extension    = ".${title}"
-    $redis_file_name_orig      = "${config_file_orig}${redis_config_extension}"
-    $redis_file_name           = "${config_file}${redis_config_extension}"
+    $redis_server_name    = "redis-server-${name}"
+    $redis_file_name_orig = sprintf('%s/%s.%s', dirname($config_file_orig), $redis_server_name, 'conf.puppet')
+    $redis_file_name      = sprintf('%s/%s.%s', dirname($config_file), $redis_server_name, 'conf')
+  }
+
+  $_real_log_file = $log_file ? {
```

Comment:
I don't think this is needed, we could simply set the log_file in the instance default declaration, then set the default to be `${log_dir}/redis-server-${name}.log`